From db3ec4f2ff5d25dbce51fa6da6c08a3d45ac5278 Mon Sep 17 00:00:00 2001
Message-Id: <db3ec4f2ff5d25dbce51fa6da6c08a3d45ac5278.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Sat, 31 Jan 2015 08:39:00 -0500
Subject: [PATCH 86/95] MUST_REBASE [VPG] : drm/i915/adf: Reinit adf hpd from
 chv power well

Since HDMI is implemented in adf driver. After resume, we need to call
adf_hpd_init for HDMI hotplug to work

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need
to work on i915 and ADF convergence path before it can be
upstreamed

Issue: GMINL-5501
Change-Id: I80f5804f8b1f107fbd78dd1afd36708cc2b9dcb9
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_adf_wrapper.h |    2 ++
 drivers/gpu/drm/i915/intel_pm.c         |    8 +++++++-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_adf_wrapper.h b/drivers/gpu/drm/i915/i915_adf_wrapper.h
index 65b4c39..abe3f4e 100644
--- a/drivers/gpu/drm/i915/i915_adf_wrapper.h
+++ b/drivers/gpu/drm/i915/i915_adf_wrapper.h
@@ -10,6 +10,8 @@ void i915_adf_wrapper_init(struct drm_i915_private *dev_priv);
 void i915_adf_wrapper_teardown(void);
 void vlv_adf_sideband_rw(struct drm_i915_private *dev_priv, u32 devfn,
 			u32 port, u32 opcode, u32 reg, u32 *val);
+extern struct intel_adf_context *adf_ctx;
+extern void intel_adf_hpd_init(struct intel_adf_context *ctx);
 #endif
 
 #endif /* __I915_ADF_WRAPPER_H__ */
diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index bf9bf66..cc10089 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -28,6 +28,7 @@
 #include <linux/cpufreq.h>
 #include "i915_drv.h"
 #include "intel_drv.h"
+#include "i915_adf_wrapper.h"
 #include "../../../platform/x86/intel_ips.h"
 #include <linux/module.h>
 #include <linux/vgaarb.h>
@@ -7015,7 +7016,12 @@ static void chv_pipe_power_well_enable(struct drm_i915_private *dev_priv,
 		if (dev_priv->power_domains.initializing)
 			return;
 
-		intel_hpd_init(dev_priv->dev);
+#ifdef CONFIG_ADF_INTEL_VLV
+		if (i915.enable_intel_adf)
+			intel_adf_hpd_init(adf_ctx);
+		else
+#endif
+			intel_hpd_init(dev_priv->dev);
 
 		i915_redisable_vga_power_on(dev_priv->dev);
 	}
-- 
1.7.9.5

